src/app/modules/home/components/user-list/user-list.component.ts

```diff
@@ -0,0 +1,279 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { MatTableDataSource } from '@angular/material/table';
+import { filter, first } from 'rxjs';
+import { MatDialog } from '@angular/material/dialog';
+import { Sort } from '@angular/material/sort';
+import { User } from '@app-models';
+import { AuthService, UserService } from '@app-services';
+import { UpdateUserComponent } from '../update-user/update-user.component';
+import { DeleteUserComponent } from '../delete-user/delete-user.component';
+import { NewUserComponent } from '../new-user/new-user.component';
+import { DeleteAllComponent } from '../delete-all/delete-all.component';
+import { Router } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'app-user-list',
+  templateUrl: './user-list.component.html',
+  styleUrls: ['./user-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class UserListComponent implements OnInit, OnChanges {
```

Comment:
Refactor needed - group some of the properties/variables into new or existing models. Topic for discussion in person.